### DISCUSSION TITLE
add_transform to model groundwater levels in wetlands

### DISCUSSION THREAD

**Discussion by adurane** [pastas/pastas#679] [Q&A] [answered]
Hello,
I am trying to model groundwater table (GWT) in a peatland in which excess water is immediately evacuated as runoff when the GWT reaches the surface. As a consequence, GWT is at ground level and almost flat in winter. For this particular project I also need to include the impact of groundwater abstraction at a nearby abstraction well, so it seems I can't use a TarsoModel (what is the reason for this, by the way?). I got some success with the following model using a ThresholdTransform:
```
ml = ps.Model(gwdata)
sm_1 = ps.StressModel(prec, rfunc=ps.FourParam(), name="prec", settings="prec")
sm_2 = ps.StressModel(evap, rfunc=ps.FourParam(), name="evap", settings="evap", up=False)
ml.add_stressmodel([sm_1, sm_2])
ml.add_transform(ps.ThresholdTransform())
ml.solve(noise=False)
ml.solve(noise=True, initial=False)
ml.plot()
plt.show()
```
![image](https://github.com/pastas/pastas/assets/98659867/efbc1721-38c3-4ff3-9cd3-158be40dd091)
However, as you can see on the above image, the modelled GWT occasionally goes significantly above the maximum observed level (i.e. the ground surface). Is there a way to improve this? Can one further constrain the ThresholdTransform transform or specify a different dampening function? 
Thanks for your help.
Arnaud

**Comment by mbakker7** [comment]
The TARSO model was specifically designed for this. Have you tried that?
https://pastas.readthedocs.io/en/latest/api/generated/generated/pastas.stressmodels.TarsoModel.html

**Reply by mbakker7** [reply]
Although I have to point out that the current Tarso can not be combined with a well.

**Reply by adurane** [reply]
Thanks for your reply. Yes, I did try the Tarso model - but as you pointed out it cannot be combined with a well (why is that?), so it can't be used for my specific project, right? And I still get simulated values well above the ground surface:
```
ml = ps.Model(gwdata, constant=False)
ts = ps.TarsoModel(prec, evap, gwdata)
ml.add_stressmodel(ts)
ml.solve(noise=False)
ml.solve(noise=True, initial=False)
ml.plot()
plt.show()
```
![image](https://github.com/pastas/pastas/assets/98659867/9cc70a09-fd3a-4a70-a8a6-d34f89b3d2c7)

**Comment by rubencalje** [comment, marked as answer]
ThresholdTransform can be seen as a system with a different storage coefficient above a certain groundwater level, for example when there is surface water above ground level in wet periods. No extra water is removed, and the response times do not change. The initial idea was that there could be multiple thresholds, but right now there can only be one threshold (nparam =2).

TarsoModel can be seen as a system with two different drainage system, where the top drainage system becomes active above a certain groundwater level. Groundwater is removed more quickly above this level, and the reponse time of your model changes. This stressmodel is more appropriate for your case. However, as TarsoModel needs to know the groundwater level, the groundwater level is calculated by the stressmodel itself, and there cannot be any other influences, like a well, in the current implementation.

So I think you have to live with the disadvantages of the stressmodel/transform you choose... Improving ThresholdTransform so there can be multiple thresholds might improve the simulated values above ground-level. This may physically not be the right solution, however.

**Reply by adurane** [reply]
This answers my questions. I am looking forward to an implementation of the TarsoModel allowing for other influences, as this would be really useful in wetland hydrology and beyond! Thanks a lot for your help and this great package.
